hscontrol/handlers.go

```diff
@@ -210,34 +206,17 @@ func (h *Headscale) RegisterWebAPI(
 		return
 	}
 
-	var content bytes.Buffer
-	if err := registerWebAPITemplate.Execute(&content, registerWebAPITemplateConfig{
+	var htmlContent = registerWebAPITemplate(registerWebAPITemplateConfig{
 		Key: machineKey.String(),
-	}); err != nil {
-		log.Error().
-			Str("func", "RegisterWebAPI").
-			Err(err).
-			Msg("Could not render register web API template")
-		writer.Header().Set("Content-Type", "text/plain; charset=utf-8")
-		writer.WriteHeader(http.StatusInternalServerError)
-		_, err = writer.Write([]byte("Could not render register web API template"))
-		if err != nil {
-			log.Error().
-				Caller().
-				Err(err).
-				Msg("Failed to write response")
-		}
-
-		return
-	}
+	})
 
 	writer.Header().Set("Content-Type", "text/html; charset=utf-8")
 	writer.WriteHeader(http.StatusOK)
-	_, err = writer.Write(content.Bytes())
-	if err != nil {
+	if _, err := writer.Write([]byte(htmlContent.Render())); err != nil {
```

Comment:
and then you can call it here like:

`if _, err := writer.Write([]byte(registerWebHTML(machineKey.String()).Render())); err != nil {`